app/migrations/Version20180502142950.php

```diff
@@ -0,0 +1,85 @@
+        $this->addSql('ALTER TABLE transaction ADD CONSTRAINT FK_723705D14DC1279C FOREIGN KEY (donation_id) REFERENCES donations (id)');
+        $this->addSql(
+            'INSERT INTO transaction (donation_id, paybox_result_code, paybox_authorization_code, paybox_payload, paybox_date_time, paybox_transaction_id, created_at)
+                    (SELECT id, paybox_result_code, paybox_authorization_code, paybox_payload,
+                      IF (paybox_payload->"$.date" IS NOT NULL AND paybox_payload->"$.time" IS NOT NULL, STR_TO_DATE(
+                        CONCAT(
+                          SUBSTR(paybox_payload->"$.date", 2, 2),
+                          \',\',
+                          SUBSTR(paybox_payload->"$.date", 4, 2),
+                          \',\',
+                          SUBSTR(paybox_payload->"$.date", 6, 4),
+                          \' \',
+                          SUBSTR(paybox_payload->"$.time", 2, 8)
+                        ),
+                        \'%d,%m,%Y %H:%i:%s\'
+                      ), NULL),
+                      IF (paybox_payload->"$.transaction" != \'0\', paybox_payload->"$.transaction", NULL),
+                      created_at
+                      FROM donations)'
+        );
+        $this->addSql('ALTER TABLE donations MODIFY status VARCHAR(25) NOT NULL, MODIFY updated_at DATETIME NOT NULL');
+        $this->addSql('ALTER TABLE donations DROP paybox_result_code, DROP paybox_authorization_code, DROP paybox_payload, DROP finished');
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE donations ADD paybox_result_code VARCHAR(100) DEFAULT NULL, ADD paybox_authorization_code VARCHAR(100) DEFAULT NULL, ADD paybox_payload JSON DEFAULT NULL COMMENT \'(DC2Type:json_array)\', ADD finished TINYINT(1) NOT NULL, CHANGE updated_at donated_at DATETIME DEFAULT NULL');
+        $this->addSql(
+            'UPDATE transaction T, donations D
+            SET
+                D.paybox_payload = T.paybox_payload,
+                D.paybox_result_code = T.paybox_result_code,
+                D.paybox_authorization_code = T.paybox_authorization_code,
+                D.finished = IF (D.status like \'%waiting_confirmation%\', 0, 1),
+                D.donated_at = IF (D.status like \'%waiting_confirmation%\', NULL, D.donated_at)
+            WHERE D.id = T.donation_id'
+        );
+        $this->addSql('ALTER TABLE donations DROP status, DROP paybox_order_ref');
+        $this->addSql('DROP TABLE transaction');
+        $this->addSql('DROP INDEX donation_uuid_idx ON donations');
+    }
+
+    public function setContainer(ContainerInterface $container = null)
```

Comment:
No migrations are NOT autoloaded, so not configured but the container, so do not use autowiring

Reply:
It's possible since Symfony 2.4 : https://symfony.com/doc/master/bundles/DoctrineMigrationsBundle/index.html#container-aware-migrations

It has already been used here https://github.com/EnMarche/en-marche.fr/blob/master/app/migrations/Version20170904153003.php#L14